File: docs/guides/python/searchWithLogicalOr.py
from algoliasearch.search.client import SearchClientSync

from algoliasearch.search.models.search_params import SearchParams


_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


query = "the query"
optional_words = ["the", "query"]
search_params = SearchParams(
    query=query,
    optional_words=optional_words,
)

_client.search_single_index(
    index_name="<YOUR_INDEX_NAME>",
    search_params=search_params,
)

File: docs/guides/go/src/searchWithLogicalOr.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func searchWithLogicalOr() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	query := "the query"
	optionalWords := search.ArrayOfStringAsOptionalWords([]string{"the", "query"})
	searchParams := search.SearchParamsObjectAsSearchParams(
		search.NewSearchParamsObject().
			SetQuery(query).
			SetOptionalWords(optionalWords),
	)

	_, err = client.SearchSingleIndex(client.NewApiSearchSingleIndexRequest(
		"<YOUR_INDEX_NAME>").WithSearchParams(searchParams))
	if err != nil {
		panic(err)
	}
}
